src/peer/util.js

```diff
@@ -177,6 +179,66 @@ function promiseTimeout<T> (timeout: number, promise: Promise<T>): Promise<T> {
   })])
 }
 
+function flatMap<T, U> (array: Array<T>, f: (x: T) => Array<U>): Array<U> {
+  return [].concat(...array.map(x => f(x)))
+}
+
+function objectIdsForQueryResult (result: QueryResultValueMsg): Array<string> {
+  let values: Array<SimpleValueMsg> = []
+
+  const simpleResultValue: ?SimpleValueMsg = _.get(result, 'simple')
+  if (simpleResultValue != null) {
+    values = [simpleResultValue]
+  } else {
+    const compoundResultBodies: ?Array<{key: string, value: SimpleValueMsg}> =
+      _.get(result, 'compound.body')
+    if (compoundResultBodies != null) {
+      values = compoundResultBodies.map(b => b.value)
+    }
+  }
+
+  const statementBodies: Array<StatementBodyMsg> = values
+    .map(v => _.get(v, 'stmt'))
+    .filter(stmt => stmt != null)
+    .map((stmt: StatementMsg) => stmt.body)
+
+  const simpleStatements: Array<SimpleStatementMsg> =
+    flatMap(statementBodies, s => {
+      const stmt: ?SimpleStatementMsg = _.get(s, 'simple')
+      if (stmt != null) return [stmt]
+
+      const compoundStatements: Array<SimpleStatementMsg> =
+        _.get(s, 'compound.body')
+      if (compoundStatements != null) return compoundStatements
+      return []
+    })
+
+  return simpleStatements.map(s => s.object)
+}
+
+function expandQueryResult (result: QueryResultValueMsg, dataObjects: Array<DataObjectMsg>): Object {
+  const objectMap = {}
+  for (const obj of dataObjects) {
+    let val: Object | string
+    try {
+      val = decode(obj.data)
+    } catch (err) {
+      val = obj.data.toString('base64')
+    }
+    objectMap[obj.key] = val
+  }
+
+  function replacer (value: any) {
+    const key = _.get(value, 'object')
+    const data = objectMap[key]
+    if (data != null) {
+      return _.set(value, 'object', {key, data})
```

Comment:
ah I guess we have all these other lodash calls too, let's take the whole package then